code/modules/paperwork/pen.dm

```diff
@@ -287,6 +342,56 @@
+	. = ..()
+	var/datum/component/transforming/transform_comp = GetComponent(/datum/component/transforming)
+	if(HAS_TRAIT(src, TRAIT_TRANSFORM_ACTIVE))
+		transform_comp.do_transform(src, user)
+	RegisterSignal(dart.loaded_projectile, COMSIG_PROJECTILE_FIRE, PROC_REF(on_containing_dart_fired))
+	RegisterSignal(dart.loaded_projectile, COMSIG_PROJECTILE_ON_SPAWN_DROP, PROC_REF(on_containing_dart_drop))
+	RegisterSignal(dart.loaded_projectile, COMSIG_PROJECTILE_ON_SPAWN_EMBEDDED, PROC_REF(on_containing_dart_embedded))
+
+/obj/item/pen/edagger/on_removed_from_dart(datum/source, obj/item/ammo_casing/dart, obj/projectile/projectile, mob/user)
+	. = ..()
+	if(istype(dart))
+		UnregisterSignal(dart, list(COMSIG_ITEM_UNEMBEDDED, COMSIG_ITEM_FAILED_EMBED))
+	UnregisterSignal(projectile, list(COMSIG_PROJECTILE_FIRE, COMSIG_PROJECTILE_ON_SPAWN_DROP, COMSIG_PROJECTILE_ON_SPAWN_EMBEDDED))
+
+/obj/item/pen/edagger/get_dart_var_modifiers(datum/source, list/modifiers)
+	. = ..()
+	var/datum/component/transforming/transform_comp = GetComponent(/datum/component/transforming)
+	modifiers["damage"] = max(5, transform_comp.throwforce_on)
+	modifiers["speed"] = max(0, transform_comp.throw_speed_on - 3)
+	var/list/embed_params = modifiers["embedding"]
+	embed_params["embed_chance"] = 100
+
+/obj/item/pen/edagger/proc/on_containing_dart_fired(obj/projectile/source)
+	SIGNAL_HANDLER
+	playsound(source, 'sound/weapons/saberon.ogg', 5, TRUE)
+	var/datum/component/transforming/transform_comp = GetComponent(/datum/component/transforming)
+	source.hitsound = transform_comp.hitsound_on
+	source.set_light(light_range, light_power, light_color, l_on = TRUE)
+
+/obj/item/pen/edagger/proc/on_containing_dart_drop(datum/source, obj/item/ammo_casing/new_casing)
+	SIGNAL_HANDLER
+	playsound(new_casing, 'sound/weapons/saberoff.ogg', 5, TRUE)
+
+/obj/item/pen/edagger/proc/on_containing_dart_embedded(datum/source, obj/item/ammo_casing/new_casing)
+	SIGNAL_HANDLER
+	RegisterSignal(new_casing, COMSIG_ITEM_UNEMBEDDED, PROC_REF(on_embedded_removed))
+	RegisterSignal(new_casing, COMSIG_ITEM_FAILED_EMBED, PROC_REF(on_containing_dart_failed_embed))
+
+/obj/item/pen/edagger/proc/on_containing_dart_failed_embed(obj/item/ammo_casing/source)
+	SIGNAL_HANDLER
+	playsound(source, 'sound/weapons/saberoff.ogg', 5, TRUE)
+	UnregisterSignal(source, list(COMSIG_ITEM_UNEMBEDDED, COMSIG_ITEM_FAILED_EMBED))
+
+/obj/item/pen/edagger/proc/on_embedded_removed(obj/item/ammo_casing/source, mob/living/carbon/victim)
+	SIGNAL_HANDLER
+	playsound(source, 'sound/weapons/saberoff.ogg', 5, TRUE)
+	UnregisterSignal(source, list(COMSIG_ITEM_UNEMBEDDED, COMSIG_ITEM_FAILED_EMBED))
+	victim.visible_message(span_warning("The blade of the [hidden_name] retracts as the [source.name] is removed from [victim]!"), span_warning("The blade of the [hidden_name] retracts as the [source.name] is removed from you!"), blind_message = span_warning("You hear an energy blade retract!"), vision_distance = 1)
```

Comment:
```suggestion
	victim.visible_message(
		span_warning("The blade of the [hidden_name] retracts as the [source.name] is removed from [victim]!"),
		span_warning("The blade of the [hidden_name] retracts as the [source.name] is removed from you!"), 		
		blind_message = span_warning("You hear an energy blade retract!"),
		vision_distance = 1,
	)
```

much cleaner. you should also name all the args here btw rather than doing the half-job.